students/ZachCooper/session05/comprehension_exercise.py

```diff
@@ -0,0 +1,118 @@
+#!usr/bin/env python3
+
+food_prefs = {"name": "Chris",
+              "city": "Seattle",
+              "cake": "chocolate",
+              "fruit": "mango",
+              "salad": "greek",
+              "pasta": "lasagna"}
+
+#1 Print the dict by passing it to a string format method
+
+print("{name} is from {city}, and he likes {cake}, {fruit} fruit, {salad} salads, and {pasta}".format(**food_prefs))
+
+#2 Using a list comprehension, build a dictionary of numbers from zero to fifteen and the hexadecimal equivalent (string is fine). 
+# (the hex() function gives you the hexidecimal representation of a number as a string)
+# Here is the LONG version:
+nums_range = range(16)
+hexes = []
+for num in nums_range:
+    hexes.append(hex(num))
```

Comment:
this is the classic use case for a comprehension:

```
hexes = [hex(num) for num in range(16)]
```
and you can actually do the dict all at once, too:

```
{num: hex(num} for num in range(16)}
```